ci-operator/step-registry/ipi/conf/aws/ipi-conf-aws-commands.sh

```diff
@@ -60,15 +63,31 @@ MAX_ZONES_COUNT="${#ZONES[@]}"
 # Save max zones count information to ${SHARED_DIR} for use in other scenarios
 echo "${MAX_ZONES_COUNT}" >> "${SHARED_DIR}/maxzonescount"
 
+existing_zones_setting=$(/tmp/yq r "${CONFIG}" 'controlPlane.platform.aws.zones')
 
-ZONES_COUNT=${ZONES_COUNT:-2}
-ZONES=("${ZONES[@]:0:${ZONES_COUNT}}")
-ZONES_STR="[ "
-ZONES_STR+=$(join_by , "${ZONES[@]}")
-ZONES_STR+=" ]"
-echo "AWS region: ${REGION} (zones: ${ZONES_STR})"
+if [[ ${existing_zones_setting} == "" ]]; then
+  ZONES_COUNT=${ZONES_COUNT:-2}
+  ZONES=("${ZONES[@]:0:${ZONES_COUNT}}")
+  ZONES_STR="[ $(join_by , "${ZONES[@]}") ]"
+  echo "AWS region: ${REGION} (zones: ${ZONES_STR})"
+  PATCH="${SHARED_DIR}/install-config-zones.yaml.patch"
+  cat >> "${PATCH}" << EOF
+controlPlane:
+  platform:
+    aws:
+      zones: ${ZONES_STR}
+compute:
+- platform:
+    aws:
+      zones: ${ZONES_STR}
+EOF
+  /tmp/yq m -x -i "${CONFIG}" "${PATCH}"
+else
+  echo "zones already set in install-config.yaml, skipped"
+fi
 
-cat >> "${CONFIG}" << EOF
+PATCH="${SHARED_DIR}/install-config-common.yaml.patch"
+cat >> "${PATCH}" << EOF
```

Comment:
I think you want the overwrite `> "${PATCH}"` instead of the append `>> "${PATCH}"` here and above.  Although apparently the magic of `yq`'s merge logic is keeping the `>>` from breaking?